code/src/App.js

```diff
@@ -1,9 +1,59 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+
+import { ThoughtInput } from 'components/ThoughtInput'
+import { ThoughtMessage } from 'components/ThoughtMessage'
 
 export const App = () => {
+
+  const [thoughts, setThoughts] = useState([])
+  const THOUGHTS_URL = 'https://happy-thoughts-technigo.herokuapp.com/thoughts'
+
+  useEffect(() => {
+    fetchThoughts()
+  }, [])
+
+  const fetchThoughts = () => {
+    /* fetch data from server, 20 latest thought posted and details for each post*/
+    fetch(THOUGHTS_URL)
+      .then((res)=>{
+          return res.json()
+      })
+      .then(data => {
+        setThoughts(data) /* data is an array of thoughts */
+      })
+  }
+
+  /*posts message and then creates a new fetch to update the site with new information from the server*/ 
+  const postThought = (thought) => {
+    fetch(THOUGHTS_URL, {
+      method: 'POST',
+      headers: { 'Content-Type':'application/json' },
+      body:JSON.stringify({message: thought})
+    })
+      .then(()=> {
+        fetchThoughts()
+      })
```

Comment:
I know that Van had a   .catch(error => console.error(error)) here, after the  fetchThoughts(),  if there are any unexpected errors in the data then we get a notice. I guess for debugging it good....could it be added maybe? I put it in my code 😬  just in case.